pkg/migrations/op_create_constraint.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package migrations
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/lib/pq"
+
+	"github.com/xataio/pgroll/pkg/db"
+	"github.com/xataio/pgroll/pkg/schema"
+)
+
+var _ Operation = (*OpCreateConstraint)(nil)
+
+func (o *OpCreateConstraint) Start(ctx context.Context, conn db.DB, latestSchema string, tr SQLTransformer, s *schema.Schema, cbs ...CallbackFn) (*schema.Table, error) {
+	table := s.GetTable(o.Table)
+
+	switch o.Type { //nolint:gocritic // more cases will be added
+	case OpCreateConstraintTypeUnique:
+		return table, o.addUniqueConstraint(ctx, conn)
+	}
+
+	return table, nil
+}
+
+func (o *OpCreateConstraint) Complete(ctx context.Context, conn db.DB, tr SQLTransformer, s *schema.Schema) error {
+	switch o.Type { //nolint:gocritic // more cases will be added
+	case OpCreateConstraintTypeUnique:
+		uniqueOp := &OpSetUnique{
+			Table: o.Table,
+			Name:  o.Name,
+		}
+		return uniqueOp.Complete(ctx, conn, tr, s)
+	}
+
+	return nil
+}
+
+func (o *OpCreateConstraint) Rollback(ctx context.Context, conn db.DB, tr SQLTransformer, s *schema.Schema) error {
```

Comment:
@andrew-farries I added a rollback operation here to remove the unique index. However, in `op_set_unique.go` the function `Rollback` is a no-op. Which one is the correct rollback process?

Reply:
`OpSetUnique.Rollback` is a no-op because the operation is a sub-operation of `OpAlterColumn`. `OpAlterColumn` handles the rollback logic for its sub-operations so there is nothing left for `OpSetUnique.Rollback` to do.

`OpAlterColumn.Rollback` performs a rollback by removing the column that was duplicated and modified along with the triggers that were added to the table.

`OpAlterColumn` contains a lot of the common logic for its various sub-operations. But as `OpCreateConstraint` is a 'standalone' operation we need to implement rollbacks here ourselves.